flink-connectors/flink-connector-kafka-base/src/test/java/org/apache/flink/streaming/connectors/kafka/table/KafkaSourceSinkFactoryTestBase.java

```diff
@@ -0,0 +1,379 @@
+		specificOffsets.put(new KafkaTopicPartition(TOPIC, PARTITION_1), OFFSET_1);
+
+		ScanFormat<DeserializationSchema<RowData>> scanFormat =
+				new TestFormatFactory.ScanFormatMock(",", true);
+
+		// Construct table source using options and table source factory
+		ObjectIdentifier objectIdentifier = ObjectIdentifier.of(
+				"default",
+				"default",
+				"scanTable");
+		CatalogTable catalogTable = createKafkaSourceCatalogTable();
+		final DynamicTableSource actualSource = FactoryUtil.createTableSource(null,
+				objectIdentifier,
+				catalogTable,
+				new Configuration(),
+				Thread.currentThread().getContextClassLoader());
+
+		// Test scan source equals
+		final KafkaScanSourceBase actualKafkaSource = (KafkaScanSourceBase) actualSource;
+		assertThat(actualKafkaSource, instanceOf(getExpectedScanSourceClass()));
+		assertEquals(producedDataType, actualKafkaSource.outputDataType);
+		assertEquals(TOPIC, actualKafkaSource.topic);
+		assertEquals(KAFKA_PROPERTIES, actualKafkaSource.properties);
+		assertEquals(specificOffsets, actualKafkaSource.specificStartupOffsets);
+		assertEquals(scanFormat, actualKafkaSource.scanFormat);
+
+		// Test Kafka consumer
+		ScanTableSource.ScanRuntimeProvider provider =
+				actualKafkaSource.getScanRuntimeProvider(ScanRuntimeProviderContext.INSTANCE);
+		assertThat(provider, instanceOf(SourceFunctionProvider.class));
+		final SourceFunctionProvider sourceFunctionProvider = (SourceFunctionProvider) provider;
+		final SourceFunction<RowData> sourceFunction = sourceFunctionProvider.createSourceFunction();
+		assertThat(sourceFunction, instanceOf(getExpectedConsumerClass()));
+	}
+
+	@Test
+	public void testTableSink() {
+		final DataType consumedDataType = SINK_SCHEMA.toPhysicalRowDataType();
+		SinkFormat<SerializationSchema<RowData>> sinkFormat =
+				new TestFormatFactory.SinkFormatMock(",");
+
+		// Construct table sink using options and table sink factory.
+		ObjectIdentifier objectIdentifier = ObjectIdentifier.of(
+				"default",
+				"default",
+				"sinkTable");
+		final CatalogTable sinkTable = createKafkaSinkCatalogTable();
+		final DynamicTableSink actualSink = FactoryUtil.createTableSink(
+				null,
+				objectIdentifier,
+				sinkTable,
+				new Configuration(),
+				Thread.currentThread().getContextClassLoader());
+
+		assertThat(actualSink, instanceOf(getExpectedSinkClass()));
+
+		assertThat(actualSink, instanceOf(KafkaSinkBase.class));
+		final KafkaSinkBase kafkaSink = (KafkaSinkBase) actualSink;
+		assertEquals(consumedDataType, kafkaSink.consumedDataType);
+		assertEquals(TOPIC, kafkaSink.topic);
+		assertEquals(KAFKA_PROPERTIES, kafkaSink.properties);
+		assertThat(kafkaSink.partitioner.isPresent(), is(true));
+		assertThat(kafkaSink.partitioner.get(), instanceOf(FlinkFixedPartitioner.class));
+
+		// Test sink format.
+		final KafkaSinkBase actualKafkaSink = (KafkaSinkBase) actualSink;
+		assertEquals(sinkFormat, actualKafkaSink.sinkFormat);
```

Comment:
Could we assert the KafakSink object ? Just like what `KafkaTableSourceSinkFactoryTestBase` do?